app.rb

```diff
@@ -0,0 +1,179 @@
+    end
+  end
+
+  def create_teacher
+    puts 'Creating teacher...'
+    name = '', age = 0, specialization = ''
+    loop do
+      print 'Name: '
+      name = gets.chomp
+      print "\n"
+      print 'Age: '
+      age = gets.chomp
+      print "\n"
+      print 'specialization: '
+      specialization = gets.chomp
+      print "\n"
+      age = age.to_i
+      break if name.length.positive? && specialization.length.positive? && age.positive?
+
+      puts 'Wrong input'
+    end
+    puts "Teacher entered.. #{name}  #{age} #{specialization}"
+    Teacher.new(specialization, age, name)
+  end
+
+  def create_student
+    puts 'Creating student...'
+    name = '', age = 0, parent_permission = false
+    loop do
+      print 'Name: '
+      name = gets.chomp
+      print "\n"
+      print 'Age: '
+      age = gets.chomp
+      age = age.to_i
+      print "\n"
+      print 'Has parent permission [Y/N]: '
+      answer = gets.chomp
+      parent_permission = true if answer.upcase == 'Y' && age > 18
+      print "\n"
+      break if name.length.positive? && [true, false].include?(parent_permission) && age.positive?
+
+      puts 'Wrong input: Try again *'
+    end
+    puts "Student entered.. #{name}  #{age} #{parent_permission}"
+    Student.new(name, age, parent_permission)
+  end
+
+  def create_person
+    loop do
+      print "\n"
+      print 'Do you want to create a student(1) or teacher(2)? [input number]: '
+      answer = gets.chomp
+      answer = answer.to_i
+      person = nil
+      case answer
+      when 1
+        person = create_student
+      when 2
+        person = create_teacher
+      else
+        puts 'Wrong input'
+      end
+      next unless person
+
+      persons.push(person)
+      puts 'Person created successfully'
+      break
+    end
+  end
+
+  def create_book
+    puts 'Creating book...'
+    title = '', author = ''
+    loop do
+      print 'Title: '
+      title = gets.chomp
+      print "\n"
+      print 'Author: '
+      author = gets.chomp
+      print "\n"
+      break if title.length.positive? && author.length.positive?
+
+      puts 'Wrong input: Try again *'
+    end
+    puts "Book entered.. #{title}  #{author}"
+    book = Book.new(title, author)
+    books.push(book)
+  end
+
+  def create_rental
+    return if persons.empty? || books.empty?
+
+    date = '', person_index = 0, book_index = 0
+    loop do
+      list_people
+      puts 'Select a person from persons listed by number: '
+      person_index = parse_int
+      list_books
+      puts 'Select a book from books listed by number: : '
+      book_index = parse_int
+      puts 'Enter date : [2022-07-24 | yy.mm.dd]'
+      date = parse_date
+      break if person_index.positive? && book_index.positive? && date
+
+      puts 'Wrong input: Try again *'
+    end
+    person = persons[person_index - 1]
+    book = books[book_index - 1]
+    rental = Rental.new(date, person, book)
+    rentals.push(rental)
+  end
+
+  def list_rentals
```

Comment:
- [ ] As the previous reviewer mentioned, the rentals returned an error and break the app after providing the person's id

![Screenshot_20220730_171807](https://user-images.githubusercontent.com/25227022/181926247-618690dc-ae9b-4031-90b0-3f3e77a48229.png)

Reply:
@Ismailco resolved issue thanks 👍🏽 
![image](https://user-images.githubusercontent.com/94127418/181934555-8c682314-b7f9-4c17-b9b0-e3cc707bf428.png)